### PR TITLE
libinput: add support for touchscreen set matrix

### PR DESCRIPTION
**Pull request by SnowNF**
My android pad emit a strange touch input from libinput, need to rotate it in 270 degree clockwise to fix.

I really like this project.
This is my first pr, I would be happy if give me some advice or other things.

### CODE REVIEW

**Review by johanmalm — COMMENTED** [review]
Thanks for contributing... and a great start :smile:

It looks good on the whole. I have left some comments on minor stuff.

Would you be happy to add a bit in `docs/labwc-config.5.scd` too?

**Comment by Consolatis** [comment]
> This is my first pr, I would be happy if give me some advice or other things.

Welcome on board then :)

As you closed your earlier PR #1644 and reopened this one, just pushing to your branch should automatically update this PR. So there is usually no need to close / reopen PRs.

**Comment by SnowNF** [comment]
> As you closed your earlier PR #1644 and reopened this one, just pushing to your branch should automatically update this PR.

I force-push my branch and it cannot be reopened. :)

File: src/seat.c [hunk 1]
```diff
@@ -237,6 +237,14 @@ configure_libinput(struct wlr_input_device *wlr_input_device)
 		wlr_log(WLR_INFO, "send events mode configured");
 		libinput_device_config_send_events_set_mode(libinput_dev, dc->send_events_mode);
 	}
+
+	if (libinput_device_config_calibration_has_matrix(libinput_dev) == 0
```
[the remark this answers is not in the document]
Comment (SnowNF) [reply src/seat.c:241]: Thanks for the advice above, I will do these later. :)

**Comment by SnowNF** [comment]
The link "https://wayland.freedesktop.org/libinput/doc/latest/absolute-axes.html#calibration-of-absolute-devices" is too long if I add to `docs/labwc-config.5.scd`. How can I fix this ?

**Comment by Consolatis** [comment]
> The link "https://wayland.freedesktop.org/libinput/doc/latest/absolute-axes.html#calibration-of-absolute-devices" is too long if I add to `docs/labwc-config.5.scd`. How can I fix this ?

It seems the issue is not the link link but rather the tables. They are detected as scd formatting syntax, this should fix it:
```diff
diff --git a/docs/labwc-config.5.scd b/docs/labwc-config.5.scd
index ee734433..629a5362 100644
--- a/docs/labwc-config.5.scd
+++ b/docs/labwc-config.5.scd
@@ -740,6 +740,7 @@ extending outward from the snapped edge.
 	matrix has no effect on relative events.
 
 	Given a 6-element array [a b c d e f], the matrix is applied as
+```
 	[ a  b  c ]   [ x ]
 	[ d  e  f ] * [ y ]
 	[ 0  0  1 ]   [ 1 ]
@@ -749,6 +750,7 @@ extending outward from the snapped edge.
 	[ 0 -1 1]		[ -1  0 1]		[  0 1 0 ] 		[ -1 0 1 ]
 	[ 1  0 0]		[  0 -1 1]		[ -1 0 1 ]		[  1 0 0 ]
 	[ 0  0 1]		[  0  0 1]		[  0 0 1 ]		[  0 0 1 ]
+```
 
 	visit https://wayland.freedesktop.org/libinput/doc/latest/absolute-axes.html#calibration-of-absolute-devices
 	for more information
```

**Comment by SnowNF** [comment]
I followed you patch, but the format in man is incorrect. I dont know how to fix this.

`
           Given a 6-element array [a b c d e f], the matrix is applied as
                [ a  b  c ]   [ x ]
                [ d  e  f ] * [ y ]
                [ 0  0  1 ]   [ 1 ]

           The most common matrices are:
                90 deg cw:          180 deg cw:         270 deg cw:         reflect along y axis:
                [ 0 -1 1]      [ -1  0 1]          [  0 1 0 ]          [ -1 0 1 ]
                [ 1  0 0]      [  0 -1 1]          [ -1 0 1 ]          [  1 0 0 ]
                [ 0  0 1]      [  0  0 1]          [  0 0 1 ]          [  0 0 1 ]
`

**Comment by Consolatis** [comment]
> I followed you patch, but the format in man is incorrect. I dont know how to fix this.

I am not that familiar with scdoc either but I assume it has something to do with the use of tabs between the boxes. Maybe it works when using spaces instead.

**Comment by SnowNF** [comment]
You are right! And now I need to do some tests on the code I just modified.

**Comment by SnowNF** [comment]
My test are all finished. Dear maintainer, could you please review my code?

**Comment by johanmalm** [comment]
Thanks. Great contribution.

**Pull request merged**